src/pki/validation.ts

```diff
@@ -0,0 +1,69 @@
+import { initCrypto } from "./crypto";
+import { bufferToCertificate } from "./utils";
+import { ValidationError } from "../validation/index";
+import { error, warning } from "../validation/utils";
+
+/**
+ * @param obj 
+ * @param property 
+ * @param issuerCertificateBinary
+ * certificate can only be verified with the issuerCertificateBinary parameter, 
+ * otherwise the verification is skipped
+ * @returns 
+ */
+export const isValidCertificate = async <T>(
```

Comment:
The `InstitutionListsIndex` also checks for the validity of the certificate. So that class should probably use this function instead of implementing it itself.
At least for the `InstitutionListsIndex`, it is necessary to add a parameter to specify the date to check for.

Probably this is related to that the `InstitutionListsIndex` uses a different dependency for certificate parsing.